ginivision/src/androidTest/java/net/gini/android/vision/test/JpegByteArraySubject.java

```diff
@@ -0,0 +1,120 @@
+package net.gini.android.vision.test;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.common.truth.FailureStrategy;
+import com.google.common.truth.Subject;
+import com.google.common.truth.SubjectFactory;
+
+import net.gini.android.vision.test.exif.ExifReader;
+import net.gini.android.vision.test.exif.ExifReaderException;
+
+class JpegByteArraySubject extends Subject<JpegByteArraySubject, byte[]> {
+
+    // Not true that <"subject"> "verb" <"expected">. "failure message".
+    private static final String RAW_MESSAGE_TEMPLATE = "Not true that <%s> %s <%s>. %s.";
+
+    static SubjectFactory<JpegByteArraySubject, byte[]> jpegByteArray() {
+        return new SubjectFactory<JpegByteArraySubject, byte[]>() {
+
+            @Override
+            public JpegByteArraySubject getSubject(final FailureStrategy fs, final byte[] that) {
+                return new JpegByteArraySubject(fs, that);
+            }
+        };
+    }
+
+    JpegByteArraySubject(@NonNull final FailureStrategy failureStrategy,
+            @Nullable final byte[] subject) {
+        super(failureStrategy, subject);
+    }
+
+    void hasUUIDinUserComment(@Nullable  String uuid) {
+        isNotNull();
+        String verb = "has in User Comment UUID";
+
+        if (uuid == null) {
+            fail(verb, (Object) null);
+            return;
+        }
+
+        final String userComment = readExifUserComment(verb, uuid, getSubject());
+        final String uuidInUserComment = getValueForKeyfromUserComment("UUID", userComment);
+
+        if (uuidInUserComment == null) {
+            failWithRawMessage(RAW_MESSAGE_TEMPLATE, getSubject(), verb, uuid,
```

Comment:
Instead of exposing `RAW_MESSAGE_TEMPLATE ` everywhere, can't you have a failure method that takes all the needed parameters? It would be easier to write and read.

Or are you concerned that the failing function name will then be incorrect?